cabal-install/Distribution/Client/ProjectPlanning/Types.hs

```diff
@@ -434,7 +434,8 @@ elabDistDirParams shared elab = DistDirParams {
             ElabPackage _ -> Nothing,
         distParamCompilerId = compilerId (pkgConfigCompiler shared),
         distParamPlatform = pkgConfigPlatform shared,
-        distParamOptimization = elabOptimization elab
+        distParamOptimization = elabOptimization elab,
+        distParamProfiling = elabProfLib elab || elabProfExe elab
```

Comment:
I am not sure if this is the proper way to check this. If anyone has a better suggestion, I’m all ears.

Reply:
Ok, here's the tricky thing...

If we are dealing with standard packages where we can split them up and handle the components individually (ie `elabPkgOrComp elab` is a `ElabComponent _`) then I think this approach should work ok, since each component is either being built with profiling or without, but not a mixture.

On the other hand if we're dealing with a multi-component package that we have to handle as a monolit (ie `elabPkgOrComp elab` is a `ElabPackage _`) then I think we have a problem because different components within that package can be built for profiling or not. For example we can enable both the profiling and non-profiling flavours of libs, and just non-profiling of an exe in the same package. It's worth remembering that the `--enable-library-profiling` and flag _adds_ additional artefacts, it doesn't switch the flavour of a single set of artefacts.